core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2137,13 +2137,14 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
           withMethodInputs(methodConfig, userInfo) { (wdl, methodInputs) =>
             withSubmissionEntityRecs(submissionRequest, workspaceContext, methodConfig.rootEntityType, dataAccess) { jobEntityRecs =>
               withWorkflowFailureMode(submissionRequest) { workflowFailureMode =>
+                val nonEmptyMethodInputs = methodInputs.filter(input => !input.workflowInput.optional || !input.expression.isEmpty)
```

Comment:
suggest `!(input.workflowInput.optional && input.expression.isEmpty)` for readability! (and confirm this is logically identical!)